### ISSUE TITLE
SophieHubert/Project_2_Mid-Bootcamp

### ISSUE THREAD

**Issue by SophieHubert** [Ironhack-Data-0621-Remote/mid-bootcamp-project#19]
https://github.com/SophieHubert/Project_2_Mid-Bootcamp.git

**Comment by SophieHubert** [comment]
https://github.com/SophieHubert/Project_2_Mid-Bootcamp.git

**Comment by Fominayasg** [comment]
# Hi Sophie 😁 !!

Let's have a look at your project. 

## README file
Is very well structured and it's so nice that you used some format tips, it makes it look much more professional.

Regarding the sections I only have a couple of suggestions.
In the ```"Main libraries used on the project"``` part, you must list all the libraries you used in the project, not only the main ones, so if anyone that doesn't know anything about Ironhack gets to your repo, he can verify wich packages to install before doing a similar project.

And if you want you can link them to their documentation writing something like this:

```
- [Pandas](https://pandas.pydata.org/docs/)
- [Matplotlib](https://matplotlib.org/)
- [Sklearn](https://scikit-learn.org/stable/#)
- [Seaborn](https://seaborn.pydata.org/)
```
That will look like this:
- [Pandas](https://pandas.pydata.org/docs/)
- [Matplotlib](https://matplotlib.org/)
- [Sklearn](https://scikit-learn.org/stable/#)
- [Seaborn](https://seaborn.pydata.org/)

The other thing I'd say about the README is that you don't need a "Documents" section where you list all the documents, because we can already see it when we open the repo. But if you want to write a section explaining a little bit the organisation of the repo, it's totally fine.
## Repo structure
The repository is quite well structured, with folders that make the reader's life easier (thanks for that :stuck_out_tongue_winking_eye:)
On the other hand I think that you have repeated files in ```"data"``` folder and ```"classification"``` folder, in my opinion it looks better if everything is in the "data" folder.

You have a couple of functions in a ```functions.ipynb``` file, it would be nice if you put them into a ```.py``` file so you can import them to other jupyter files and reuse them with copying the code.


I loved that you numbered the ```.ipnyb``` files :heart_eyes:.

## SQL

#### Question 2
On this question you had to create a new table, that is exactly the same that the one that you already had. Yes, it sounds a little silly but is just to make sure that you know how to create a new table and assign each column the correct dtype. You could do something like this:
```sql
drop table if exists credit_card_data;
create table credit_card_data  (
	Customer_Number	int,
    Offer_Accepted varchar(20),
	Reward varchar(20),  
	Mailer_Type varchar(20),
    Income_Level varchar(20),
    Bank_Accounts_Open int,
    Overdraft_Protection varchar(20) ,
    Credit_Rating varchar(20),
    Credit_Cards_Held int,
    Homes_Owned int,
    Household_Size int,
    Own_Your_Home varchar(20),
    Average_Balance float,
    Q1_Balance float,
    Q2_Balance float,
    Q3_Balance float,
    Q4_Balance float,
    primary key (Customer_Number)
    );
```

#### Question 5
In this one I think you forgot to drop the column. It should be like this.
```sql
alter table credit_card_data
drop column Q4_Balance;

select * from credit_card_data
limit 10;
```

#### Question 11
This question is perfect! I only want to give you a tip. In the part of the query where yo write:
```sql
where credit_rating = 'medium' or credit_rating = 'high'
```
you can have the same result writing this:
```sql
where Credit_Rating <> "Low"
```
maybe another time can be useful

#### Question 17
The way you answer this question works but you have to count the 11th position manually, to have only the information of this customer you can do this.

```sql
select * from credit_card_data 
order by Q1_balance asc 
limit 10,1;
```

On question 13 it says that there's an error on the syntax, make sure every column name is written without typos etc...

Every other question are perfect. Really good job with the SQL part Sophie!!
## EDA & Data cleaning
You did a good job using dictionaries to rename columns. Maybe you could have got a little deeper in the data to know if you needed to clean something else like categoricl variables using .uniques() or any other idea you can think about.

## Data preprocessing
You did a good preprocessing  of the data dealing with categorical variables and rescaling others. Well done!
## ML
About the models, the KNN is well done but a linear regression doesn't make sense i na classification project, but I think you already realized that because you only developed more in depth the KNN model. 

If you have time maybe you can try with some other classification models to see if the are any other that works better for your data.
## Tableau
The Tableau part was missing or may be i haven't been able to find it. If you had any problems with the licence using tableau you can use any other visuaization tools. 

To sum up, you did a good job with the project. Keep on going!!
